tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/parser/PathVisitor.java

```diff
@@ -64,12 +64,28 @@ private String parseNodeName(PathParser.NodeNameContext ctx) {
     if (nodeName.startsWith(TsFileConstant.BACK_QUOTE_STRING)
         && nodeName.endsWith(TsFileConstant.BACK_QUOTE_STRING)) {
       String unWrapped = nodeName.substring(1, nodeName.length() - 1);
-      if (StringUtils.isNumeric(unWrapped)
+      if (isRealNumber(unWrapped)
           || !TsFileConstant.IDENTIFIER_PATTERN.matcher(unWrapped).matches()) {
         return nodeName;
       }
       return unWrapped;
     }
     return nodeName;
   }
+
+  /** Return true if the str is a real number. Examples: 1.0; +1.0; -1.0; 0011; 011e3; +23e-3 */
+  public static boolean isRealNumber(String str) {
+    if (str.startsWith("+")) {
+      return NumberUtils.isCreatable(str.substring(1));
+    }
+    int index = 0;
+    // remove zeros
+    for (int i = 0, n = str.length(); i < n; i++) {
+      if (str.charAt(i) != '0') {
+        index = i;
+        break;
+      }
+    }
+    return NumberUtils.isCreatable(str.substring(index));
+  }
```

Comment:
duplicated with previous, and it seems that `+0001` and `-00001` won't be considered

Reply:
Fixed.